File: StaffManagement/AddShiftViewController.m
//
//  AddShiftViewController.m
//  StaffManagement
//
//  Created by Chris Jones on 2017-03-13.
//  Copyright © 2017 Derek Harasen. All rights reserved.
//

#import "AddShiftViewController.h"

@interface AddShiftViewController ()

@end

@implementation AddShiftViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationController.title = @"Set Up Shift";
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)addShift:(UIBarButtonItem *)sender {
    
}

- (IBAction)cancelAddShift:(UIBarButtonItem *)sender {
    [self.navigationController popViewControllerAnimated:YES];
}


/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: StaffManagement/AddShiftViewController.swift
//
//  AddShiftViewController.swift
//  StaffManagement
//
//  Created by Stephen Gilroy on 2016-06-12.
//  Copyright © 2016 Derek Harasen. All rights reserved.
//

import UIKit

class AddShiftViewController: UIViewController {
    
    var waiter: Waiter!

    @IBOutlet var startDatePicker: UIDatePicker!
    @IBOutlet var endDatePicker: UIDatePicker!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    // MARK: - Button Functions
    @IBAction func addButton(sender: UIButton) {
        
        //Add new shift to Core Data
        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        let managedContext = appDelegate.managedObjectContext
        
        let entity =  NSEntityDescription.entityForName("Shift", inManagedObjectContext: managedContext)
        
        let newShift = NSManagedObject(entity: entity!, insertIntoManagedObjectContext: managedContext)
        let startDate: NSDate = startDatePicker.date
        let endDate: NSDate = endDatePicker.date
        
        newShift.setValue(self.waiter, forKey: "waiter")
        newShift.setValue(startDate, forKey: "start")
        newShift.setValue(endDate, forKey: "end")
        
        //Save context
        do {
            try managedContext.save()
        } catch let error as NSError  {
            print("Could not save \(error), \(error.userInfo)")
        }
        
        //Dismiss to previous view
        dismissViewControllerAnimated(true, completion: nil)
    }
    
    // MARK: - Dismiss to previous view
    @IBAction func backButton(sender: UIButton) {
        dismissViewControllerAnimated(true, completion: nil)
    }

}
